### PR TITLE
feat: automated testing

### PR DESCRIPTION
**Pull request by mihai-sysbio**
### Main improvements in this PR:
To simplify development, there should be a way for tests to run automatically. This PR aims to resolve this by adding basic support to run the currently defined tests.

This PR is based on a version of `devel` that has the latest from the main branch, which is why there is long list of commits. I found this to be the easiest way of pulling in the `.github` folder that was needed, which seemed to exist only on the master branch.

**I hereby confirm that I have:**

- [ ] Tested my code on my own machine
- [x] Followed the [development guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/DevGuidelines).
- [x] Selected `devel` as a target branch
- [ ] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about this PR

### CODE REVIEW

**Comment by mihai-sysbio** [comment]
Maybe it would be more useful if the test results would be added as a comment to a PR?

**Comment by edkerk** [comment]
The reason why all the `updateVersion` commits are only applied to `main` and not `devel` (the reason for the long list of commits here), is that `main` contains `version.txt` which is updated with each release. `devel` does not have this file, to avoid confusion that the status of `devel` is not representative of a particular release (except just after it has been pushed to `main`).

Meanwhile, I don't see any strong reason why `.github` is only in `main`. I suggest:
- Make a separate PR to "copy" the `.github` from `main` to `devel`
  - During the next PR to `main` this might results in a conflict, but should be easily resolved
- Reapply 9f153e59111773c434dbc601db1f6d97066ac10b to a new feat-branch that is based on the updated `devel`.

These two steps can actually be combined as well.

**Comment by mihai-sysbio** [comment]
> But yes, might be more _accessible_ to directly see the test results. However, what if further commits are added to the same PR, does it leave a comment with each test run?

Indeed, but we could make it conditional, i.e. only on test failures.

File: .github/issue_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+INSERT CODE HERE
+```
+
+#### System information
+* Please report:
+1. RAVEN version (stabile [release](https://github.com/SysBioChalmers/RAVEN/releases), `devel` branch?)
+2. Operating system (Windows/Mac/Linux; include version)
+
+**I hereby confirm that I have:**
+<!-- Note: replace [ ] with [X] to check the box. PLEASE DELETE THIS LINE -->
```
Comment (mihai-sysbio) [inline .github/issue_template.md:20]:
```suggestion
<!-- Note: replace [ ] with [X] to check the box -->
```

**Review by edkerk — APPROVED** [review]

**Comment by edkerk** [comment]
As you suggested above, the GitHub Action can post a comment with the MATLAB command line output?

**Comment by mihai-sysbio** [comment]
@edkerk the PR commenting has been implemented in this PR.
I've been trying to split the output on multiple lines, but didn't quite get that to work. I'm hoping it's acceptable for now.
If anyone has any ideas, feel free to try them directly on this branch.

**Review by edkerk — COMMENTED** [review]
Looks very nice! But now only the first line is shown. Perhaps better to show all results, even if it is not split over separate lines? Reverting the last commit will do this?

**Comment by mihai-sysbio** [comment]
> Looks very nice! But now only the first line is shown. Perhaps better to show all results, even if it is not split over separate lines? Reverting the last commit will do this?

Yes, I removed the last commit and the PR comment was consequently updated.

**Review by edkerk — APPROVED** [review]

**Pull request merged**